File: backend/insurance_engine/InsuranceEngine/InsuranceEngine.Domain/Entities/FieldViolation.cs
using System.ComponentModel.DataAnnotations;

namespace InsuranceEngine.Domain.Entities;

public class FieldViolation
{
    [Key]
    public Guid FieldViolationId { get; set; }

    [Required]
    [MaxLength(200)]
    public string Field { get; set; } = string.Empty;

    [Required]
    [MaxLength(100)]
    public string Code { get; set; } = string.Empty;

    [Required]
    [MaxLength(1000)]
    public string Description { get; set; } = string.Empty;

    [Required]
    [MaxLength(500)]
    public string RejectedValue { get; set; } = string.Empty;

    [Required]
    public Guid ErrorId { get; set; }

    public Error Error { get; set; } = null!;
}

File: sdks/insuretech-go-sdk/pkg/models/field_violation.go
package models


// FieldViolation represents a field_violation
type FieldViolation struct {
	Code string `json:"code,omitempty"`
	Description string `json:"description,omitempty"`
	Field string `json:"field,omitempty"`
	RejectedValue string `json:"rejected_value,omitempty"`
}
